soa-derive-internal/src/input.rs

```diff
@@ -230,4 +250,38 @@ impl Input {
     pub fn ptr_mut_name(&self) -> Ident {
         Ident::new(&format!("{}PtrMut", self.name), Span::call_site())
     }
-}
+    pub fn fold_fields(&self, 
```

Comment:
Could you add a small documentation comment to this method? It looks like it is performing a fold over the fields, but why this is needed instead of `self.fields.iter().fold(...)` is not immediately clear to me

Reply:
Replaced by a combination of `self.iter_fields().map(|(ident, ty, is_nested)| /* generate the tokenstreams */).concat_by(...)`

`concat_by` comes from a `TokenStreamIterator` trait, which is convenient than direct `fold`, I add some documentations and unit test about new methods.